lib/ContentHelpers.php

```diff
@@ -206,4 +207,43 @@ public static function getFundingProgrammeThumbnailUrlNew($entry)
             'crop' => 'faces',
         ]) : null;
     }
+
+    // Returns a set of open graph meta tags, optionally matching a ?social=<slug> querystring
+    public static function extractSocialMetaTags(Entry $entry)
+    {
+        $openGraph = [];
+
+        if ($entry->socialMediaTags) {
+            $allTagSets = $entry->socialMediaTags->all();
+
+            // Default to using the first set of tags
+            $ogData = $entry->socialMediaTags->one();
+
+            // If we've passed a ?social=<slug> parameter, try to find its matching
+            // set of tags (eg. for per-URL open graph metadata)
+            if ($searchQuery = \Craft::$app->request->getParam('social')) {
+                $ogSlugs = array_column($allTagSets, 'ogSlug');
```

Comment:
Does Craft give us any finder methods to do this, rather than calling `->all()` can you perform a query?